saleor/account/models.py

```diff
@@ -120,7 +120,7 @@ class User(PermissionsMixin, AbstractBaseUser):
     default_billing_address = models.ForeignKey(
         Address, related_name='+', null=True, blank=True,
         on_delete=models.SET_NULL)
-
+    email_change_requested_on = models.DateTimeField(null=True)
```

Comment:
Do we want to keep a track on when was the last email change?

If that's just for the error purpose, we could read the email back from cache and determine if the time passed or not